CMakeLists.txt

```diff
@@ -109,38 +110,23 @@ add_custom_target(check-all)
 add_custom_target(docs ALL)
 add_dependencies(docs mlir-doc)
 
-option(AIR_BUILD_RUNTIME "Build the air runtime" ON)
-if (NOT LibXAIE_FOUND)
-	message(WARNING "LibXAIE not found, AIR runtime will not be built.")
-	set(AIR_BUILD_RUNTIME OFF)
-endif()
-
-if(CMAKE_USE_TOOLCHAIN_AIRHOST)
-	set(AIRHOST_TOOLCHAIN_OPT ${CMAKE_TOOLCHAIN_FILE_OPT})
-else()
-	set(AIRHOST_TOOLCHAIN_OPT ${CMAKE_TOOLCHAIN_FILE})
-endif() 
-
-if(AIR_BUILD_RUNTIME)
-	message("AIRHOST_TOOLCHAIN_OPT=${AIRHOST_TOOLCHAIN_OPT}")
-	message("Sysroot=${Sysroot}")
-	ExternalProject_Add(air_runtime_lib
-		PREFIX ${CMAKE_CURRENT_BINARY_DIR}/runtime_libTmp
+foreach(target ${AIR_RUNTIME_TARGETS})
+	message("Compiling AIR runtimes for ${target}")
+	message("TOOLCHAIN=${${target}_TOOLCHAIN_FILE}")
+	message("Sysroot=${${target}_Sysroot}")
+	ExternalProject_Add(air_runtime_lib_${target}
+		PREFIX ${CMAKE_CURRENT_BINARY_DIR}/runtime_libTmp/${target}
 		SOURCE_DIR ${PROJECT_SOURCE_DIR}/runtime_lib
-		BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/runtime_lib
+		BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/runtime_lib/${target}
 		CMAKE_ARGS
 			-DCMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}
-			-DCMAKE_TOOLCHAIN_FILE=${AIRHOST_TOOLCHAIN_OPT}
-			-DArch=${Arch}
-			-DgccVer=${gccVer}
-			-DSysroot=${Sysroot} 
+			-DCMAKE_TOOLCHAIN_FILE=${${target}_TOOLCHAIN_FILE}
+			-DSysroot=${${target}_Sysroot} 
 			-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
 			-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
 			-DCMAKE_ASM_COMPILER=${CMAKE_ASM_COMPILER}
 			-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
 			-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
-			-Dpybind11_DIR=${pybind11_DIR}
-			-DLibXAIE_DIR=${LibXAIE_DIR}
 			-DAIE_DIR=${AIE_DIR}
 			-DBUILD_AIR_PCIE=${BUILD_AIR_PCIE}
```

Comment:
Should these go in the target toolchain file?  That seems like the only place to pass different parameters depending on the target without adding a bunch of ad-hoc logic in the foreach loop.

Reply:
Perhaps... it wasn't clear to me whether these are per-target options?

Reply:
Today BUILD_AIR_PCIE is roughly synonymous with !aarch64 in the runtime code, but not quite. We really have 3 sub-targets: build with PCIe support, build with /dev/mem support, and build with /dev/mem support but insert extra error messages because you're not building for aarch64. The latter is unused but would be useful for unit-testing parts of the runtime or as an emulation runtime.

Reply:
It's not clear to me what exactly to do with this.   Any suggestions?

Reply:
I like the idea of using toolchain files per target. It would make sense and enable outside-AMD users. Especially because they cannot use type 2. Rather than per board, it is per system: PCIe, embedded, hybrid.

Reply:
I'm not jgmelber, but my thoughts are that platforms we want to support now and in the future are will be split between the more embedded ARM+AIE, and then the data centerish x86 + (ARM+AIE), so I do think figuring this out for the current gen is useful, even if the result is proliferation to begin with